### ISSUE TITLE
Add Dowty's proto-roles to perceptual generation

### ISSUE THREAD

**Issue by rgabbard** [isi-vista/adam#104]

**Comment by rgabbard** [comment]
@denizbeser : Dowty's paper is http://linguistics.berkeley.edu/~syntax-circle/syntax-group/dowty91.pdf ; we are interested in (27) and (28) in p. 572.

@charles-yang-upenn  @mitchmarcus : I think we don't want to include the proto-roles in the perception directly, but rather than finer-grained properties (a)-(d) under each of (27) and (28) (and assume the learner module uses these to form proto-roles).  Do you agree?

@denizbeser : Pending comment from Mitch and Charles, proceed with making perceivable properties for all of (a) - (d) under (27) and (28).  Then add an `asserted_properties`  field to `ActionDescription` which should be a map from `SituationObject` to a collection of `OntologyNode`s for properties we want to say it has if this action takes place.  Finally, alter perception generation to apply these properties.

**Comment by rgabbard** [comment]
@denizbeser : example for `PUT`:

```python
_PUT_ACTION_DESCRIPTION = ActionDescription(
    frame=ActionDescriptionFrame({AGENT: _PUT_AGENT, THEME: _PUT_THEME, GOAL: _PUT_GOAL}),

    ....
    asserted_properties = [
    (_PUT_AGENT, VOLITIONAL_INVOLVMENT),
    (_PUT_AGENT, CAUSES_CHANGE_OF_STATE),
    (_PUT_THEME, CHANGE_OF_STATE),
)
```

I think there are more we could perhaps add, but that gives you the idea.  I am actually a bit unsure how the `GOAL` of `put` should be handled; I see [here](http://web.stanford.edu/~bclevin/dowty19fin.pdf) that: 
> Some researchers recast additional traditional semantic roles as proto-roles. Primus
(1999:54–55, 137–138) introduces a Recipient Proto-role to deal with phenomena involving ditransitive verbs, benefactives, and addressees. This proto-role combines Proto-Agent
and Proto-Patient properties.

@charles-yang-upenn  @mitchmarcus : If either of you has time today and would like to pitch in, I'm sure @denizbeser would appreciate help in assigning the proto-role sub-properties to semantic roles of the various actions in the vocabulary.

**Comment by mitchmarcus** [comment]
I would simply leave the GOAL of put as a location, and hence as a GOAL.  After looking at the Levin paper you cite (and I trust Beth's intuitions a _lot_), I think that multiplying proto-roles until we absolutely have to is not the right way to go.  Every new proto-role will involve learning new inferences, which we may be forced into with increasing vocabularly, but we should keep it simple (as kids evidently do!) until forced.  Otherwise, we'll ending addressing lots of difficulties by coming up with rare proto-roles.  From the discussion of linguists on corner cases here, it's clear that there's a set of things that aren't clear, but they require a consistent story across as many cases as possible.  Also, many linguists want to simply collapse Dowty's properties into a couple of very abstract ones, as linguists and physicists love to do.  I think we should resist this for a while - we need less inferences to learn, not more, and I think lots of linguists overdo it.

**Comment by rgabbard** [comment]
@mitchmarcus : That sounds good. I think the only thing to note is that in the assignment of the sub-proto-role atomic properties, we need to consider also objects in the scene outside the core roles. For example, for `PUT`, the thing the object is placed on, although not the `PATIENT` or `THEME`, nonetheless is `STATIONARY_WITH_RESPECT_TO_ACTION`.  We don't want to make the learner's job too easy. :-)

**Comment by rgabbard** [comment]
Closed by #274
